### ISSUE TITLE
unused_braces false positive for macro expression.

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#88104]
I tried this code:

```rust
macro_rules! foo(
    ($imp: expr) => {
        pub fn f() -> i32 {
            $imp
            1
        }
    }
);

foo!(
    { println!("hi") }
);

fn main() {}
```

I expected to see this happen: No warning.

Instead, this happened: `unused_braces` warning with a suggestion that causes the code to fail to compile.

```
warning: unnecessary braces around block return value
  --> src/main.rs:11:5
   |
11 |     { println!("hi") }
   |     ^^^^^^^^^^^^^^^^^^ help: remove these braces
   |
   = note: `#[warn(unused_braces)]` on by default
```

After applying the suggestion:

```
error: expected `;`, found `1`
  --> src/main.rs:4:17
   |
4  |               $imp
   |                   ^ help: add `;` here
5  |               1
   |               - unexpected token
...
10 | / foo!(
11 | |     println!("hi")
12 | | );
   | |__- in this macro invocation
   |
   = note: this error originates in the macro `foo` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Found when trying to run `cargo fix` on [`nphysics2d`](https://crates.io/crates/nphysics2d) 0.24.0.

### Meta

`rustc --version --verbose`:
```
rustc 1.56.0-nightly (0035d9dce 2021-08-16)
binary: rustc
commit-hash: 0035d9dcecee49d1f7349932bfa52c05a6f83641
commit-date: 2021-08-16
host: x86_64-apple-darwin
release: 1.56.0-nightly
LLVM version: 12.0.1
```

**Comment by bugadani** [comment]
This issue also triggers when the matcher is a `tt`, but only if the expression is in the same line as the braces. I.e. the following triggers the warning:

```rust
macro_rules! trigger_warning {
    ($tokens:tt) => {{ $tokens }}
}

pub fn main() {
    trigger_warning!({ println!("hello") });
}
```

but this does not:

```rust
macro_rules! trigger_warning {
    ($tokens:tt) => {{ $tokens }}
}

pub fn main() {
    trigger_warning!({
        println!("hello") });
}
```
